test/browser/lifecycle.js

```diff
@@ -1,6 +1,16 @@
-import { h, render, rerender, Component } from '../../src/preact';
+import { h, render, options, Component } from '../../src/preact';
 /** @jsx h */
 
+// Wrapper rerender function that fully drains the queue
+const renderQueue = [];
+options.debounceRendering = (callback) => renderQueue.push(callback);
```

Comment:
For the current render queue implementation in preact any render operations enqueued while the render queue is being drained will be dispatched in a separate microtask.

Before verifying the final state of the DOM the tests need all of the render operations (regardless of how/when they're performed) to be processed. It seemed cleaner to build a simple render queue than to attempt to guess at how many of the standard `rerender` calls would have been necessary.

It may be more correct for preact to flush all render events in a single microtask. This would be a simple adjustment, but I think such a change deserves separate discussion.

Reply:
Thanks! That was very helpful